File: Leetcode/cpp/Two Sum.cc
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        mp.clear();
        for(int i=0;i<(int)numbers.size();i++)
        {
            int x = numbers[i];
            mp[x]=i;
        }
        for(int i=0;i<(int)numbers.size();i++)
        {
            int x = numbers[i];
            int y = target - x;

            if(mp.find(y)!=mp.end())
            {
                vector<int> ans;
                ans.push_back(i+1);
                ans.push_back(mp[y]+1);
                return ans;
            }
        }
    }
    map<int, int> mp;
};

File: LintCode/python/Two Sum.py
class Solution:
    """
    @param: numbers: An array of Integer
    @param: target: target = numbers[index1] + numbers[index2]
    @return: [index1 + 1, index2 + 1] (index1 < index2)
    """
    def twoSum(self, numbers, target):
        d = {}
        for i, num in enumerate(numbers):
            if target - num in d:
                return [ d[target - num] + 1, i + 1 ]
            d[num] = i
        return [ -1, -1 ]
